arch/x86/core/pcie.c

```diff
@@ -197,7 +199,9 @@ uint32_t pcie_msi_map(unsigned int irq,
 #endif
 #endif
 	{
+		uint32_t apic_id = x86_read_loapic(LOAPIC_ID) >> 24;
 		map = 0xFEE00000U; /* standard delivery to BSP local APIC */
+		map |= (((apic_id & DEST_ID_MASK) << 12) | REDIR_HINT);
```

Comment:
Ideally enable_msix() PCI host driver's API should provide support for user/device driver to specify Interrupt Affinity since MSI support affinity per device and MSI-X provide per vector for Intel SOCs (via MSI data register).  If we able to update PCI Host interface with this should be great and I can provide another API says "pcie_msi_set_cpu_affinity()" and can assign these flag accordingly. Please share your suggestion ?

Reply:
Right, we don't have a decent solution for targetting MSI interrupts currently, we need to do some design work.

I was just asking why you were changing this.  Is it needed for the fix?  Why?

Reply:
I will create another commit for adding new MSI API for setting affinity and hence removed the changes